public/app/plugins/panel/graph/graph.ts

```diff
@@ -193,7 +194,9 @@ module.directive('grafanaGraph', function($rootScope, timeSrv) {
             continue;
           }
 
-          if (data[i].stats.timeStep < min) {
+          if (((panel.bars && (typeof data[i].bars === 'undefined' || typeof data[i].bars.show === 'undefined' || data[i].bars.show)) ||
```

Comment:
this looks a bit too complex, is there any way we can simplify this, or if not break it up? maybe add some kind of function  showAsBars() on TimeSeries class